test/gateway/eip20_cogateway/progress_mint_with_proof.js

```diff
@@ -0,0 +1,530 @@
+    );
+
+  });
+
+  it('should fail when rlp of parent nodes is zero', async function () {
+
+    let rlpParentNodes = "0x";
+    await Utils.expectRevert(
+      testEIP20CoGateway.progressMintWithProof.call(
+        messageHash,
+        rlpParentNodes,
+        params.blockNumber,
+        MessageStatusEnum.Declared,
+        {from: facilitator},
+      ),
+      'RLP parent nodes must not be zero',
+    );
+
+  });
+
+  it('should fail when storage root for block height is not set', async function () {
+
+    let blockHeight = new BN(1);
+    await Utils.expectRevert(
+      testEIP20CoGateway.progressMintWithProof.call(
+        messageHash,
+        params.serializedProof,
+        blockHeight,
+        MessageStatusEnum.Declared,
+        {from: facilitator},
+      ),
+      'Storage root must not be zero',
+    );
+
+  });
+
+  it('should fail when message status is declared revocation', async function () {
+
+    await testEIP20CoGateway.setInboxStatus(
+      messageHash,
+      MessageStatusEnum.DeclaredRevocation,
+    );
+
+    await Utils.expectRevert(
+      testEIP20CoGateway.progressMintWithProof.call(
+        messageHash,
+        params.serializedProof,
+        params.blockNumber,
+        MessageStatusEnum.Declared,
+        {from: facilitator},
+      ),
+      'Message on target must be Declared.',
+    );
+
+  });
+
+  it('should fail when message status is revoked', async function () {
+
+    await testEIP20CoGateway.setInboxStatus(
+      messageHash,
+      MessageStatusEnum.Revoked,
+    );
+
+    await Utils.expectRevert(
+      testEIP20CoGateway.progressMintWithProof.call(
+        messageHash,
+        params.serializedProof,
+        params.blockNumber,
+        MessageStatusEnum.Declared,
+        {from: facilitator},
+      ),
+      'Message on target must be Declared.',
+    );
+
+  });
+
+  it('should fail when message status is undeclared', async function () {
+
+    await testEIP20CoGateway.setInboxStatus(
+      messageHash,
+      MessageStatusEnum.Undeclared,
+    );
+
+    await Utils.expectRevert(
+      testEIP20CoGateway.progressMintWithProof.call(
+        messageHash,
+        params.serializedProof,
+        params.blockNumber,
+        MessageStatusEnum.Declared,
+        {from: facilitator},
+      ),
+      'Message on target must be Declared.',
+    );
+
+  });
+
+  it('should fail when message status at source is DeclaredRevocation', async function () {
+
+    await Utils.expectRevert(
+      testEIP20CoGateway.progressMintWithProof.call(
+        messageHash,
+        params.serializedProof,
+        params.blockNumber,
+        MessageStatusEnum.DeclaredRevocation,
+        {from: facilitator},
+      ),
+      'Message on source must be Declared or Progressed.',
+    );
+
+  });
+
+  it('should fail when message status at source is Revoked', async function () {
+
+    await Utils.expectRevert(
+      testEIP20CoGateway.progressMintWithProof.call(
```

Comment:
```suggestion
      testEIP20CoGateway.progressMintWithProof(
```

Please remove call at other places as-well.